File: SDKs/ndhm.gov.in/ndhm-hiu/0.5/go/pkg/models/shared/hiusubscriptionrequestnotificationacknowledgement.go
package shared

import (
	"time"
)

type HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnum string

const (
	HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnumOk HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnum = "OK"
)

type HiuSubscriptionRequestNotificationAcknowledgementAcknowledgement struct {
	Status                HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnum `json:"status"`
	SubscriptionRequestID string                                                                     `json:"subscriptionRequestId"`
}

type HiuSubscriptionRequestNotificationAcknowledgement struct {
	Acknowledgement *HiuSubscriptionRequestNotificationAcknowledgementAcknowledgement `json:"acknowledgement,omitempty"`
	Error           *Error                                                            `json:"error,omitempty"`
	RequestID       string                                                            `json:"requestId"`
	Resp            RequestReference                                                  `json:"resp"`
	Timestamp       time.Time                                                         `json:"timestamp"`
}

File: SDKs/ndhm.gov.in/ndhm-gateway/0.5/typescript/src/sdk/models/shared/hiusubscriptionrequestnotificationacknowledgement.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import { Error } from "./error";
import { RequestReference } from "./requestreference";

export enum HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnum {
    Ok = "OK"
}


export class HiuSubscriptionRequestNotificationAcknowledgementAcknowledgement extends SpeakeasyBase {
  @Metadata({ data: "json, name=status" })
  status: HiuSubscriptionRequestNotificationAcknowledgementAcknowledgementStatusEnum;

  @Metadata({ data: "json, name=subscriptionRequestId" })
  subscriptionRequestId: string;
}


export class HiuSubscriptionRequestNotificationAcknowledgement extends SpeakeasyBase {
  @Metadata({ data: "json, name=acknowledgement" })
  acknowledgement?: HiuSubscriptionRequestNotificationAcknowledgementAcknowledgement;

  @Metadata({ data: "json, name=error" })
  error?: Error;

  @Metadata({ data: "json, name=requestId" })
  requestId: string;

  @Metadata({ data: "json, name=resp" })
  resp: RequestReference;

  @Metadata({ data: "json, name=timestamp" })
  timestamp: Date;
}
